pkg/controllers/vdb/init_db.go

```diff
@@ -269,12 +274,70 @@ func (g *GenericDatabaseInitializer) setKerberosAuthParms() {
 	g.ConfigurationParams.Set("KerberosEnableKeytabPermissionCheck", "0")
 }
 
+func (g *GenericDatabaseInitializer) setAuthFromGCSSecret(ctx context.Context) (string, ctrl.Result, error) {
+	name := g.Vdb.Spec.Communal.CredentialSecret
+	client, err := gsm.NewClient(ctx)
+	if err != nil {
+		return "", ctrl.Result{}, err
+	}
+	defer client.Close()
+
+	req := &secretmanagerpb.AccessSecretVersionRequest{
+		Name: name,
+	}
+
+	result, err := client.AccessSecretVersion(ctx, req)
+	if err != nil {
+		g.Log.Error(err, "could not fetch secrets, credential error")
+		return "", ctrl.Result{}, err
+	}
+
+	crc32c := crc32.MakeTable(crc32.Castagnoli)
+	checksum := int64(crc32.Checksum(result.Payload.Data, crc32c))
+	if checksum != *result.Payload.DataCrc32C {
+		g.Log.Error(err, "Permission denied to read the secret")
+		return "", ctrl.Result{}, err
+	}
+
+	GcpCred := map[string]string{}
+
+	err = json.Unmarshal([]byte(string(result.Payload.Data)), &GcpCred)
+	if err != nil {
+		return "", ctrl.Result{}, fmt.Errorf("checksum failure, expected %d but got %d", *result.Payload.DataCrc32C, checksum)
```

Comment:
The error text does make sense here. Is this a copy+paste error?

Reply:
Moved it already